### PR TITLE
fix(server): set `unsafe-eval` CSP in dev mode with custom CSP policies

### PR DESCRIPTION
**Pull request by FloEdelmann**
## Types of changes

- [X] Bug fix (a non-breaking change which fixes an issue)
- [ ] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Description

This could relate to #7451. When the `render.csp.policies` config is an object, `unsafe-eval` is not added to the CSP header in development mode, like it is when an array or nothing is provided.

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes
- [X] All new and existing tests are passing.

### CODE REVIEW

**Comment by P4sca1** [comment]
Why is unsafe-eval needed in dev but not in production?

**Comment by FloEdelmann** [comment]
@P4sca1 Actually, I don't know, but it's the same way it happens in https://github.com/FloEdelmann/nuxt.js/blob/8933d375e9d2e082f2f23b491fe8241762fe4080/packages/server/src/middleware/nuxt.js#L129.

File: packages/server/src/middleware/nuxt.js [hunk 1]
```diff
@@ -134,20 +134,24 @@ const getCspString = ({ cspScriptSrcHashes, allowedSources, policies, isDev, isR
   }
 
   if (policyObjectAvailable) {
-    const transformedPolicyObject = transformPolicyObject(policies, cspScriptSrcHashes)
+    const transformedPolicyObject = transformPolicyObject(policies, cspScriptSrcHashes, isDev)
```
Comment (clarkdo) [inline packages/server/src/middleware/nuxt.js:137]: What do you think adding `'unsafe-eval'` to `cspScriptSrcHashes` if it doesn't exist in dev mode ? Then it should work for both string and object cases.

**Comment by clarkdo** [comment]
@P4sca1 It's because eval is only in dev mode because of webpack devtool config https://webpack.js.org/configuration/devtool/  and we should avoid unsafe eval code in produciton.

**Review by clarkdo — COMMENTED** [review]
Sorry, I just checked the original issue in comment, I think root cause might be something else. I'll look into it now, please hold this pr.

**Comment by clarkdo** [comment]
@FloEdelmann This issue has been fixed in v2.13.0, so this pr can be closed, we'll do some code cleanup shortly https://github.com/nuxt/nuxt.js/issues/7451#issuecomment-653124094

**Pull request closed without merging**